### PR TITLE
[RouteCollection] Add name parameter on addCollection method

### PR DESCRIPTION
**Pull request by ad3n**
| Q | A |
| --- | --- |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | no |
| License | MIT |

Add support for

``` php
<?php

$routeCollection->addCollection($collection1, 'name');
```

### CODE REVIEW

File: src/Symfony/Component/Routing/RouteCollection.php [hunk 1]
```diff
@@ -122,16 +122,17 @@ public function remove($name)
...
-    public function addCollection(RouteCollection $collection)
+    public function addCollection(RouteCollection $collection, $name = '')
     {
         // we need to remove all routes with the same names first because just replacing them
         // would not place the new route at the end of the merged array
-        foreach ($collection->all() as $name => $route) {
-            unset($this->routes[$name]);
-            $this->routes[$name] = $route;
+        foreach ($collection->all() as $oldName => $route) {
+            unset($this->routes[$oldName]);
```
Comment (Tobion) [inline src/Symfony/Component/Routing/RouteCollection.php]: it doesnt make sense to delete the old name, what you want is to delete the new name before adding it

File: src/Symfony/Component/Routing/RouteCollection.php [hunk 2]
```diff
@@ -122,16 +122,17 @@ public function remove($name)
...
+            $this->routes[sprintf('%s_%s', $oldName, $name)] = $route;
```
Comment (Tobion) [inline src/Symfony/Component/Routing/RouteCollection.php]: huge BC break currently as it now always adds an underscore

Comment (ihsanudin) [inline src/Symfony/Component/Routing/RouteCollection.php]: ok i will resolve the problem. thanks for review

**Comment by Tobion** [comment]
I'm :-1: on this one.
1. it would require to also expose an option like `name_prefix` in YAML, XML. otherwise this feature cannot be used from there
2. you can still have collision in route names if you don't craft your prefixes carefully. so after all you have to take care of duplicate route names anyway.

File: src/Symfony/Component/Routing/RouteCollection.php [hunk 3]
```diff
@@ -122,15 +122,21 @@ public function remove($name)
...
      *
      * @api
      */
-    public function addCollection(RouteCollection $collection)
+    public function addCollection(RouteCollection $collection, $newName = null)
     {
         // we need to remove all routes with the same names first because just replacing them
         // would not place the new route at the end of the merged array
         foreach ($collection->all() as $name => $route) {
             unset($this->routes[$name]);
```
Comment (Tobion) [inline src/Symfony/Component/Routing/RouteCollection.php]: this is still wrong. it need to remove the new name if there is one (`'%s_%s`)

**Comment by Tobion** [comment]
Closing for the reasons given above.
Also this is not a bugfix.

**Comment by Tobion** [comment]
Ref original ticket #13654

**Pull request closed without merging**